src/zarr/core/metadata/v3.py

```diff
@@ -70,6 +71,27 @@ def parse_dimension_names(data: None | Iterable[str | None]) -> tuple[str | None
         raise TypeError(msg)
 
 
+def parse_storage_transformers(data: object) -> tuple[dict[str, JSON], ...]:
+    if data is None:
+        return ()
+    if isinstance(data, Iterable):
+        if len(tuple(data)) >= 1:
+            msg = (
+                "Got a non-null storage_transformers keyword argument. "
+                "Storage transformers are not supported by zarr-python at this time. "
+                "The storage transformer(s) will be retained in array metadata but will not "
+                "influence storage routines"
+            )
+            warnings.warn(msg, UserWarning, stacklevel=1)
```

Comment:
after thinking about this some more, I believe we should raise an error here instead of warning. 

We allow:
- metadata w/o the storage_transformers key
- metadata with `storage_transformers == [] or None`

We error for:
- metadata w/ `len(storage_transformers) > 0`

Reply:
why not error when a user tries to create an array with the metadata? I'm thinking about how someone would develop a storage transformer with zarr-python as a dependency. If the metadata parsing functionality supports storage transformers, then they can use that immediately, and then subclass / implement their own array class that can use the storage transformer. Seems like a better workflow than forcing them to also subclass the metadata class?